### ISSUE TITLE
False positive in ThrowsCount rule

### ISSUE THREAD

**Issue by t-kameyama** [detekt/detekt#3222]
```kotlin
import java.io.IOException

fun foo(x: Int) {
    fun bar(x: Int) {
        when (x) {
            1 -> throw IOException()
            2 -> throw IOException()
            3 -> throw IOException()
        }
    }
    return bar(x)
}
```

`foo` at line 3 should not be reported as well as ReturnCount rule.
